applications/FluidDynamicsApplication/custom_utilities/fluid_auxiliary_utilities.cpp

```diff
@@ -418,4 +421,75 @@ void FluidAuxiliaryUtilities::CalculateSplitConditionGeometryData<false>(
     rpModShapeFunc->ComputeNegativeExteriorFaceAreaNormals(rNormals, FaceId, GeometryData::IntegrationMethod::GI_GAUSS_2);
 }
 
+    void FluidAuxiliaryUtilities::EmbeddedSkinToVolumeMapping(
+        ModelPart& rVolumeModelPart,
+        ModelPart& rSkinModelPart,
+        const double SearchRadius
+        )
+    {
+        //put the nodes in the skin into a search database
+        typedef BinsDynamic<3, Node<3>, std::vector<Node<3>::Pointer>> DynamicBins;
+        DynamicBins locator(rSkinModelPart.Nodes().GetContainer().begin(), rSkinModelPart.Nodes().GetContainer().end());
+
+        //reset the EMBEDDED_VELOCITY
+        block_for_each(rVolumeModelPart.Nodes(), [](auto& rNode){
+            rNode.GetValue(EMBEDDED_VELOCITY).clear();
+        });
+
+        //detect all the nodes that belong to cut elements
+        ModelPart::NodesContainerType cut_elem_nodes;
+        for(const auto& rElem : rVolumeModelPart.Elements()){
+            const auto& geom = rElem.GetGeometry();
+            
+            unsigned int npos=0, nneg=0;
+            for(unsigned int i=0; i<geom.size(); ++i){
+                if(geom[i].FastGetSolutionStepValue(DISTANCE)>0)
+                    npos++;
+                else
+                    nneg++;
+            }
+
+            if(npos>0 && nneg>0){ //element is split
+                for(unsigned int i=0; i<geom.size(); ++i)
+                    cut_elem_nodes.push_back(geom(i));
+            }
+        }
+        cut_elem_nodes.Unique();
+
+        unsigned int MaxResults = 100; //deliberately low, we do not want to interpolate from more than this nodes
+        auto TLS = std::make_pair(std::vector<Node<3>::Pointer>(MaxResults), std::vector<double>(MaxResults));
+        
+        //for every interface node (nodes in cut elements)
+        block_for_each(cut_elem_nodes, TLS, [&locator, SearchRadius](auto& rNode, auto& rTLS){
+                //find all neighbours within a radius
+                auto& neighbours = rTLS.first;
+                auto& distances  = rTLS.second;
+                const unsigned int nfound = locator.SearchInRadius( rNode, 
+                                                                SearchRadius, 
+                                                                neighbours.begin(), 
+                                                                distances.begin(), 
+                                                                neighbours.size());
+                if(nfound == 0)
+                    std::cout << "not found any neighbour!!! for node " << rNode.Id() << std::endl;
```

Comment:
Is this a leftover from debugging? If not, I guess it should go in the logger.

Reply:
ok, i transformed it into an error